### PR TITLE
JAMES-1966 upgrade commons-io to 2.5 and fix encoding warnings

### PR DESCRIPTION
**Pull request by mbaechler**
I will add many small commits to check everything is ok with CI at each step

### CODE REVIEW

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
```
[Building Java 8 james-1966-8424767f-225f-43bf-be79-2db29c16a177] Tests in error: 
[Building Java 8 james-1966-8424767f-225f-43bf-be79-2db29c16a177]   TestDiscardProtocol.testComplete:95 »  Timed out wait for be notified that rea...
```

**Comment by chibenwa** [comment]
Is there a PB with this code ?

```
[Building Java 8 james-1966-8c034bf2-0f28-403d-b68e-bb5cc4fd4bac] Tests in error: 
[Building Java 8 james-1966-8c034bf2-0f28-403d-b68e-bb5cc4fd4bac]   AbstractDomainListPrivateMethodsTest.setDefaultDomainShouldCreateFromHostnameWhenEqualsToLocalhost:100 » NoClassDefFound
[Building Java 8 james-1966-8c034bf2-0f28-403d-b68e-bb5cc4fd4bac]   AbstractDomainListPrivateMethodsTest.setDefaultDomainShouldNotCreateTwiceWhenCallingTwoTimes:112 » NoClassDefFound
[Building Java 8 james-1966-8c034bf2-0f28-403d-b68e-bb5cc4fd4bac]   AbstractDomainListPrivateMethodsTest.setDefaultDomainShouldSetFromConfigurationWhenDifferentFromLocalhost:76 » NoClassDefFound
[Building Java 8 james-1966-8c034bf2-0f28-403d-b68e-bb5cc4fd4bac]   AbstractDomainListPrivateMethodsTest.setDefaultDomainShouldSetFromHostnameWhenEqualsToLocalhost:88 » NoClassDefFound
```

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
```
[Building Java 8 james-1966-68bf449b-4620-4049-8c8f-656e70da9f7a] Results :
[Building Java 8 james-1966-68bf449b-4620-4049-8c8f-656e70da9f7a] 
[Building Java 8 james-1966-68bf449b-4620-4049-8c8f-656e70da9f7a] Tests in error: 
[Building Java 8 james-1966-68bf449b-4620-4049-8c8f-656e70da9f7a]   AbstractDomainListPrivateMethodsTest.setDefaultDomainShouldCreateFromHostnameWhenEqualsToLocalhost:100 » NoClassDefFound
[Building Java 8 james-1966-68bf449b-4620-4049-8c8f-656e70da9f7a]   AbstractDomainListPrivateMethodsTest.setDefaultDomainShouldNotCreateTwiceWhenCallingTwoTimes:112 » NoClassDefFound
[Building Java 8 james-1966-68bf449b-4620-4049-8c8f-656e70da9f7a]   AbstractDomainListPrivateMethodsTest.setDefaultDomainShouldSetFromConfigurationWhenDifferentFromLocalhost:76 » NoClassDefFound
[Building Java 8 james-1966-68bf449b-4620-4049-8c8f-656e70da9f7a]   AbstractDomainListPrivateMethodsTest.setDefaultDomainShouldSetFromHostnameWhenEqualsToLocalhost:88 » NoClassDefFound
[Building Java 8 james-1966-68bf449b-4620-4049-8c8f-656e70da9f7a] 
[Building Java 8 james-1966-68bf449b-4620-4049-8c8f-656e70da9f7a] Tests run: 61, Failures: 0, Errors: 4, Skipped: 0
```

Same error. Sorry, but you broke tests....

File: backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/CassandraCluster.java [hunk 1]
```diff
@@ -58,7 +59,7 @@ private CassandraCluster(CassandraModule module, EmbeddedCassandra embeddedCassa
             session = new FunctionRunnerWithRetry(MAX_RETRY).executeAndRetrieveObject(CassandraCluster.this::tryInitializeSession);
             typesProvider = new CassandraTypesProvider(module, session);
         } catch (Exception exception) {
-            Throwables.propagate(exception);
+            new RuntimeException(exception);
```
Comment (rouazana) [inline backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/CassandraCluster.java]: missing throw

File: backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/CassandraCluster.java [hunk 2]
```diff
@@ -100,7 +101,7 @@ private void sleep(long sleepMs) {
         try {
             Thread.sleep(sleepMs);
         } catch(InterruptedException interruptedException) {
-            Throwables.propagate(interruptedException);
+            new RuntimeException(interruptedException);
```
Comment (rouazana) [inline backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/CassandraCluster.java]: missing throw

File: backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/EmbeddedCassandra.java [hunk 3]
```diff
@@ -38,7 +36,7 @@ private EmbeddedCassandra() {
         try {
             EmbeddedCassandraServerHelper.startEmbeddedCassandra(TimeUnit.SECONDS.toMillis(20));
         } catch (ConfigurationException | TTransportException | IOException | InterruptedException e) {
-            Throwables.propagate(e);
+            new RuntimeException(e);
```
Comment (rouazana) [inline backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/EmbeddedCassandra.java]: missing throw

File: mailbox/jpa/src/test/java/org/apache/james/mailbox/jpa/mail/TransactionalMessageMapper.java [hunk 4]
```diff
@@ -102,7 +101,7 @@ public void runVoid() throws MailboxException {
                 }
             });
         } catch (MailboxException e) {
-            Throwables.propagate(e);
+            new RuntimeException(e);
```
Comment (rouazana) [inline mailbox/jpa/src/test/java/org/apache/james/mailbox/jpa/mail/TransactionalMessageMapper.java]: missing throw

File: mailbox/memory/src/test/java/org/apache/james/mailbox/inmemory/MessageIdManagerTestSystemProvider.java [hunk 5]
```diff
@@ -66,7 +64,7 @@ private static InMemoryMailboxManager createMailboxManager() {
         try {
             mailboxManager.init();
         } catch (MailboxException e) {
-            Throwables.propagate(e);
+            new RuntimeException(e);
```
Comment (rouazana) [inline mailbox/memory/src/test/java/org/apache/james/mailbox/inmemory/MessageIdManagerTestSystemProvider.java]: missing throw

File: mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/ListMessageAssert.java [hunk 6]
```diff
@@ -47,7 +46,7 @@ public InnerMessage apply(MailboxMessage input) {
                     return new InnerMessage(input.getMessageId(), input.getUid(), input.getMailboxId(), input.getInternalDate(), input.getBodyOctets(), 
                             input.getFullContentOctets(), input.getMediaType(), input.getSubType(), IOUtils.toString(input.getFullContent()));
                 } catch (IOException e) {
-                    Throwables.propagate(e);
+                    new RuntimeException(e);
```
Comment (rouazana) [inline mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/ListMessageAssert.java]: missing throw

File: mpt/impl/imap-mailbox/cyrus/src/test/java/org/apache/james/mpt/imapmailbox/cyrus/host/CyrusHostSystem.java [hunk 7]
```diff
@@ -126,7 +125,7 @@ public void executeProtocolSession(ProtocolSession protocolSession) {
                 session.stop();
             }
         } catch (Exception e) {
-            Throwables.propagate(e);
+            new RuntimeException(e);
```
Comment (rouazana) [inline mpt/impl/imap-mailbox/cyrus/src/test/java/org/apache/james/mpt/imapmailbox/cyrus/host/CyrusHostSystem.java]: missing throw

File: mpt/impl/imap-mailbox/cyrus/src/test/java/org/apache/james/mpt/imapmailbox/cyrus/host/Docker.java [hunk 8]
```diff
@@ -90,9 +89,9 @@ public void stop(ContainerCreation container) {
             dockerClient.killContainer(container.id());
             dockerClient.removeContainer(container.id(), true);
         } catch (DockerException e) {
-            Throwables.propagate(e);
+            new RuntimeException(e);
```
Comment (rouazana) [inline mpt/impl/imap-mailbox/cyrus/src/test/java/org/apache/james/mpt/imapmailbox/cyrus/host/Docker.java]: missing throw

File: mpt/impl/imap-mailbox/cyrus/src/test/java/org/apache/james/mpt/imapmailbox/cyrus/host/Docker.java [hunk 9]
```diff
@@ -90,9 +89,9 @@ public void stop(ContainerCreation container) {
...
         } catch (InterruptedException e) {
-            Throwables.propagate(e);
+            new RuntimeException(e);
```
Comment (rouazana) [inline mpt/impl/imap-mailbox/cyrus/src/test/java/org/apache/james/mpt/imapmailbox/cyrus/host/Docker.java]: missing throw

File: server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/data/CassandraDomainListModule.java [hunk 10]
```diff
@@ -68,7 +67,7 @@ public void initModule() {
                 cassandraDomainList.setLog(LOGGER);
                 cassandraDomainList.configure(configurationProvider.getConfiguration("domainlist"));
             } catch (ConfigurationException e) {
-                Throwables.propagate(e);
+                new RuntimeException(e);
```
Comment (rouazana) [inline server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/data/CassandraDomainListModule.java]: missing throw

File: server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/data/CassandraRecipientRewriteTableModule.java [hunk 11]
```diff
@@ -70,7 +69,7 @@ public void initModule() {
                 recipientRewriteTable.setLog(LOGGER);
                 recipientRewriteTable.configure(configurationProvider.getConfiguration("recipientrewritetable"));
             } catch (ConfigurationException e) {
-                Throwables.propagate(e);
+                new RuntimeException(e);
```
Comment (rouazana) [inline server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/data/CassandraRecipientRewriteTableModule.java]: missing throw

File: server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/data/CassandraUsersRepositoryModule.java [hunk 12]
```diff
@@ -70,7 +69,7 @@ public void initModule() {
                 usersRepository.setLog(LOGGER);
                 usersRepository.configure(configurationProvider.getConfiguration("usersrepository"));
             } catch (ConfigurationException e) {
-                Throwables.propagate(e);
+                new RuntimeException(e);
```
Comment (rouazana) [inline server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/data/CassandraUsersRepositoryModule.java]: missing throw

File: server/container/guice/cassandra-ldap-guice/src/main/java/org/apache/james/data/LdapUsersRepositoryModule.java [hunk 13]
```diff
@@ -67,7 +66,7 @@ public void initModule() {
                 usersRepository.configure(configurationProvider.getConfiguration("usersrepository"));
                 usersRepository.init();
             } catch (Exception e) {
-                Throwables.propagate(e);
+                new RuntimeException(e);
```
Comment (rouazana) [inline server/container/guice/cassandra-ldap-guice/src/main/java/org/apache/james/data/LdapUsersRepositoryModule.java]: missing throw

File: server/container/guice/cassandra-ldap-guice/src/test/java/org/apache/james/CassandraLdapJamesServerTest.java [hunk 14]
```diff
@@ -62,7 +60,7 @@ protected void clean() {
         try {
             imapClient.disconnect();
         } catch (IOException e) {
-            Throwables.propagate(e);
+            new RuntimeException(e);
```
Comment (rouazana) [inline server/container/guice/cassandra-ldap-guice/src/test/java/org/apache/james/CassandraLdapJamesServerTest.java]: missing throw

File: server/container/guice/guice-common/src/main/java/org/apache/james/modules/server/DNSServiceModule.java [hunk 15]
```diff
@@ -66,7 +65,7 @@ public void initModule() {
                 dnsService.configure(configurationProvider.getConfiguration("dnsservice"));
                 dnsService.init();
             } catch (Exception e) {
-                Throwables.propagate(e);
+                new RuntimeException(e);
```
Comment (rouazana) [inline server/container/guice/guice-common/src/main/java/org/apache/james/modules/server/DNSServiceModule.java]: missing throw

File: server/container/guice/guice-common/src/main/java/org/apache/james/modules/server/JMXServerModule.java [hunk 16]
```diff
@@ -115,7 +114,7 @@ public void initModule() {
                 jmxServer.register(JMX_COMPONENT_MAILBOXCOPIER, mailboxCopierManagementMBean);
                 jmxServer.register(JMX_COMPONENT_REINDEXER, reIndexerManagementMBean);
             } catch (Exception e) {
-                Throwables.propagate(e);
+                new RuntimeException(e);
```
Comment (rouazana) [inline server/container/guice/guice-common/src/main/java/org/apache/james/modules/server/JMXServerModule.java]: missing throw... OK they are too many, so please fix them all ^^

File: server/container/guice/cassandra-ldap-guice/src/test/java/org/apache/james/CassandraLdapJamesServerTest.java (hunk 14 above)
Comment (mbaechler) [inline server/container/guice/cassandra-ldap-guice/src/test/java/org/apache/james/CassandraLdapJamesServerTest.java]: sorry for these mistakes

**Comment by mbaechler** [comment]
test this please

**Comment by mbaechler** [comment]
```
initializeTypesShouldNotFailIfCalledTwice(org.apache.james.backends.cassandra.init.CassandraTypeProviderTest)  Time elapsed: 0.14 sec  <<< ERROR!
java.lang.NoClassDefFoundError: com/google/common/util/concurrent/FutureFallback
	at org.apache.james.backends.cassandra.init.CassandraTypeProviderTest.setUp(CassandraTypeProviderTest.java:66)
Caused by: java.lang.ClassNotFoundException: com.google.common.util.concurrent.FutureFallback
	at org.apache.james.backends.cassandra.init.CassandraTypeProviderTest.setUp(CassandraTypeProviderTest.java:66)
```

**Comment by chibenwa** [comment]
```
[Building Java 8 james-1966-c9dae24b-9bce-4e7c-824d-a4ced7e04aba] java.lang.NoClassDefFoundError: org/apache/commons/logging/Log
[Building Java 8 james-1966-c9dae24b-9bce-4e7c-824d-a4ced7e04aba] 	at org.apache.james.user.ldap.ReadOnlyUsersLDAPRepositoryTest.ldapRepositoryConfiguration(ReadOnlyUsersLDAPRepositoryTest.java:88)
[Building Java 8 james-1966-c9dae24b-9bce-4e7c-824d-a4ced7e04aba] 	at org.apache.james.user.ldap.ReadOnlyUsersLDAPRepositoryTest.configureShouldThrowOnNonBooleanValueForSupportsVirtualHosting(ReadOnlyUsersLDAPRepositoryTest.java:76)
[Building Java 8 james-1966-c9dae24b-9bce-4e7c-824d-a4ced7e04aba] Caused by: java.lang.ClassNotFoundException: org.apache.commons.logging.Log
[Building Java 8 james-1966-c9dae24b-9bce-4e7c-824d-a4ced7e04aba] 	at org.apache.james.user.ldap.ReadOnlyUsersLDAPRepositoryTest.ldapRepositoryConfiguration(ReadOnlyUsersLDAPRepositoryTest.java:88)
[Building Java 8 james-1966-c9dae24b-9bce-4e7c-824d-a4ced7e04aba] 	at org.apache.james.user.ldap.ReadOnlyUsersLDAPRepositoryTest.configureShouldThrowOnNonBooleanValueForSupportsVirtualHosting(ReadOnlyUsersLDAPRepositoryTest.java:76)
[Building Java 8 james-1966-c9dae24b-9bce-4e7c-824d-a4ced7e04aba] 
[Building Java 8 james-1966-c9dae24b-9bce-4e7c-824d-a4ced7e04aba] 
[Building Java 8 james-1966-c9dae24b-9bce-4e7c-824d-a4ced7e04aba] Results :
[Building Java 8 james-1966-c9dae24b-9bce-4e7c-824d-a4ced7e04aba] 
[Building Java 8 james-1966-c9dae24b-9bce-4e7c-824d-a4ced7e04aba] Tests in error: 
[Building Java 8 james-1966-c9dae24b-9bce-4e7c-824d-a4ced7e04aba]   ReadOnlyUsersLDAPRepositoryTest.configureShouldThrowOnNonBooleanValueForSupportsVirtualHosting:76->ldapRepositoryConfiguration:88 NoClassDefFound
[Building Java 8 james-1966-c9dae24b-9bce-4e7c-824d-a4ced7e04aba]   ReadOnlyUsersLDAPRepositoryTest.supportVirtualHostingShouldReturnFalseByDefault:45->ldapRepositoryConfiguration:88 NoClassDefFound
[Building Java 8 james-1966-c9dae24b-9bce-4e7c-824d-a4ced7e04aba]   ReadOnlyUsersLDAPRepositoryTest.supportVirtualHostingShouldReturnFalseWhenReportedInConfig:64->ldapRepositoryConfiguration:88 NoClassDefFound
[Building Java 8 james-1966-c9dae24b-9bce-4e7c-824d-a4ced7e04aba]   ReadOnlyUsersLDAPRepositoryTest.supportVirtualHostingShouldReturnTrueWhenReportedInConfig:52->ldapRepositoryConfiguration:88 NoClassDefFound
```

🎉

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**